noir-projects/aztec-nr/aztec/src/contract_self.nr

```diff
@@ -0,0 +1,239 @@
+use crate::{
```

Comment:
I generated the docs with AI here (other than the msg_sender funcs where they were copied) but I think the result is surprisingly good and not AI-sloppy (just did minor edits to it).

Think the verbosity here is justified as macros in general are quite cryptic and since `ContractSelf` is becoming a very important piece of Aztec.nr.